configs/src/types.ts

```diff
@@ -12,5 +14,39 @@ export type NetworkAliases = {
 };
 
 export type NetworkConfigs<T = NetworkConfig> = {
-  [network: string]: T;
+  [network in SupportedNetworks]: T;
+};
+
+export enum SupportedNetworks {
+  MAINNET = 'mainnet',
+  GOERLI = 'goerli',
+  SEPOLIA = 'sepolia',
+  POLYGON = 'polygon',
+  MUMBAI = 'mumbai',
+  BASE = 'baseMainnet',
+  BASE_GOERLI = 'baseGoerli',
+  BASE_SEPOLIA = 'baseSepolia',
+  ARBITRUM = 'arbitrum',
+  ARBITRUM_SEPOLIA = 'arbitrumSepolia',
+}
+
+// the entries in this enum has to be in order from
+// oldest to newest so that getLatestNetworkVersion() works as expected
+export enum SupportedVerions {
+  V100 = 'v1.0.0',
+  V130 = 'v1.3.0',
```

Comment:
We must be able to distinguish, for example, version 1.23.4 from 1.2.34.
```suggestion
  V1_0_0 = 'v1.0.0',
  V1_3_0 = 'v1.3.0',
```
although it looks ugly.

Reply:
valid point and I wanted it look less ugly but you are right. Changed